app/src/main/java/net/squanchy/eventdetails/EventDetailsActivity.kt

```diff
@@ -0,0 +1,141 @@
+package net.squanchy.eventdetails
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.Toolbar
+import android.view.Menu
+import android.view.MenuItem
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import net.squanchy.R
+import net.squanchy.eventdetails.widget.EventDetailsCoordinatorLayout
+import net.squanchy.navigation.Navigator
+import net.squanchy.notification.NotificationsIntentService
+import net.squanchy.schedule.domain.view.Event
+import net.squanchy.speaker.domain.view.Speaker
+
+class EventDetailsActivity : AppCompatActivity() {
+
+    private val subscriptions = CompositeDisposable()
+
+    private lateinit var service: EventDetailsService
+    private lateinit var coordinatorLayout: EventDetailsCoordinatorLayout
+
+    private lateinit var navigator: Navigator
+    private lateinit var eventId: String
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.activity_event_details)
+
+        setupToolbar()
+
+        with(eventDetailsComponent(this)) {
+            service = service()
+            navigator = navigator()
+        }
+
+        coordinatorLayout = findViewById(R.id.event_details_root)
+    }
+
+    private fun setupToolbar() {
+        val toolbar = findViewById<Toolbar>(R.id.toolbar)
+        setSupportActionBar(toolbar)
+        supportActionBar!!.setDisplayShowTitleEnabled(false)
+        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        eventId = intent.getStringExtra(EXTRA_EVENT_ID)
+
+        subscribeToEvent(eventId)
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
+        if (requestCode == REQUEST_CODE_SIGNIN) {
+            subscribeToEvent(eventId)
+        } else {
+            super.onActivityResult(requestCode, resultCode, data)
+        }
+    }
+
+    private fun subscribeToEvent(eventId: String) {
+        subscriptions.add(
+                service.event(eventId)
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe { event -> coordinatorLayout.updateWith(event, onEventDetailsClickListener(event)) }
+        )
+    }
+
+    private fun onEventDetailsClickListener(event: Event): EventDetailsCoordinatorLayout.OnEventDetailsClickListener {
```

Comment:
Why not using the expression syntax? `private fun onEventDetailsClickListener(event: Event): EventDetailsCoordinatorLayout.OnEventDetailsClickListener = object ...`

Reply:
Mostly because it's still a 10-ish line function, but I'll update it